metabase/metabase
# show columns added after the dashcard has been created in the viz settings

closes https://github.com/metabase/metabase/issues/25592

### Description
When you edit a dashcard and sort some columns, and after that you add a column to the original query, the new column is not visible in the settings.

To fix I added a getValue to the column settings that merges the dataset columns to the current settings if needed.

### How to verify

From linked issue:

> 1. create a question (new->question->people), save it and add it to a dashboard
> 2. now edit the viz settings of the card and move some columns around, save the dashboard
> 3. now go to the question and add a custom column (like concat(name, email)), save the question
> 4. go to the dashboard, you won't see the custom column, not either on the custom viz settings
> 5. reset to defaults the card, now you'll see the column appear

Using a sql query could be faster and it allows to test it also with a object detail viz:
- make a sql query suck as `select 1 as "one", 2 as "two"`
- save it to a dashboard
- hide a column from the dashboard
- change the query to be `select 1 as "one", 2 as "two", 3 as "three" and save it
- go to dashboard and you should see the new column as disabled in the settings


### Demo

https://github.com/metabase/metabase/assets/1914270/ac4b0942-b551-4d0b-857e-68afacc915aa


### Checklist
